services/mongodbflex/wait.go

```diff
@@ -0,0 +1,107 @@
+package mongodbflex
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/utils"
+	"github.com/stackitcloud/stackit-sdk-go/core/wait"
+)
+
+const (
+	InstanceStateEmpty       = ""
+	InstanceStateProgressing = "PROCESSING"
+	InstanceStateUnknown     = "UNKNOWN"
+	InstanceStateSuccess     = "READY"
+	InstanceStateFailed      = "FAILED"
+)
+
+// Interface needed for tests
+type APIClientInstanceInterface interface {
+	GetInstanceExecute(ctx context.Context, projectId, instanceId string) (*GetInstanceResponse, error)
+}
+
+func handleError(reqErr error) (res interface{}, done bool, err error) {
+	oapiErr, ok := reqErr.(*GenericOpenAPIError) //nolint:errorlint //complaining that error.As should be used to catch wrapped errors, but this error should not be wrapped
+	if !ok {
+		return nil, false, fmt.Errorf("could not convert error to GenericOpenApiError")
+	}
+	// Some APIs may return temporary errors and the request should be retried
+	if utils.Contains(wait.RetryHttpErrorStatusCodes, oapiErr.statusCode) {
+		return nil, false, nil
+	}
+	return nil, false, reqErr
+}
+
+// CreateInstanceWaitHandler will wait for creation
+func CreateInstanceWaitHandler(ctx context.Context, a APIClientInstanceInterface, projectId, instanceId string) *wait.Handler {
+	return wait.New(func() (res interface{}, done bool, err error) {
+		s, err := a.GetInstanceExecute(ctx, projectId, instanceId)
+		if err != nil {
+			return handleError(err)
+		}
+		if s == nil || s.Item == nil || s.Item.Id == nil || *s.Item.Id != instanceId || s.Item.Status == nil {
```

Comment:
I took it from the postgres flex waiter, do you know if i was a special case there?

Reply:
@hcsa73 I think it was you that implemented the waiter for postgresflex, do you remember something?

Reply:
There was no special case, I was just being overzealous. 

Same with checking if Id is nil, which can't happen because it's stated as a required field

Reply:
Right, I guess we can do the same here then